server/client.go

```diff
@@ -612,6 +612,20 @@ func (c *client) RegisterNkeyUser(user *NkeyUser) error {
 	return nil
 }
 
+func splitSubject(subject string) (string, string) {
+	vals := strings.Split(subject, " ")
+	var s, q string
+	n := len(vals)
+	if n == 1 {
+		s = vals[0]
+	} else if n >= 2 {
```

Comment:
Will we catch "foo bar baz" as an error or just silently do "foo bar" essentially?

Reply:
We should catch the error earlier in config parsing and options validation I guess.

Reply:
Yes we should change to catch these errors earlier now...

Reply:
I made the split function return an error. In `setPermissions` we handle it by logging and dropping that permission.

This would be in addition to handling it in the config parsing and the JWT code.

https://github.com/nats-io/nats-server/pull/1143/commits/d4d71e990d6065bb6aa50240219e27fd51e63e25